File: examples/CSharp/LicenseValidator.cs
using System;
using System.IO;
using System.Security.Cryptography;
using System.Text;
using Newtonsoft.Json;
using Newtonsoft.Json.Linq;

public class LicenseValidator
{
    private readonly ILogger _logger;
    private readonly HardwareFingerprint _fingerprint;

    public LicenseValidator()
    {
        _logger = new Logger();
        _fingerprint = new HardwareFingerprint();
    }

    public ValidationResult ValidateLicense(string licenseFile, string licenseKey, bool validateFootprint = true)
    {
        try
        {
            // 加载并解密许可证文件
            string encryptedLicense = File.ReadAllText(licenseFile);
            if (string.IsNullOrEmpty(encryptedLicense))
            {
                return ValidationFailed("无法加载许可证文件");
            }

            string decryptedLicense = Decrypt(encryptedLicense, licenseKey);
            var licenseData = JsonConvert.DeserializeObject<JObject>(decryptedLicense);

            // 生成当前硬件指纹
            string currentHardwareFingerprint = _fingerprint.GenerateFingerprint();
            if (validateFootprint && licenseData["hardware_fingerprint"].ToString() != currentHardwareFingerprint)
            {
                return ValidationFailed("硬件指纹不匹配");
            }

            // 检查许可证是否过期
            long currentTime = DateTimeOffset.UtcNow.ToUnixTimeSeconds();
            long expiresAt = licenseData["expires_at"].ToObject<long>();
            long issuedAt = licenseData["issued_at"].ToObject<long>();

            if (currentTime > expiresAt || currentTime < issuedAt)
            {
                return ValidationFailed("许可证已过期");
            }

            var result = new ValidationResult
            {
                Valid = true,
                Expiry = expiresAt,
                Features = licenseData["features"].ToObject<string[]>(),
                MaxUsers = licenseData["max_users"].ToObject<int>()
            };

            _logger.Info($"许可证验证成功，详情: {JsonConvert.SerializeObject(result)}");
            return result;
        }
        catch (Exception e)
        {
            _logger.Error($"许可证验证错误: {e.Message}");
            return ValidationFailed(e.Message);
        }
    }

    private string Decrypt(string encryptedText, string key)
    {
        // 解密逻辑
        return "dummy_decrypted_license";
    }

    private ValidationResult ValidationFailed(string message)
    {
        return new ValidationResult
        {
            Valid = false,
            Message = message
        };
    }
}

public class ValidationResult
{
    public bool Valid { get; set; }
    public long Expiry { get; set; }
    public string[] Features { get; set; }
    public int MaxUsers { get; set; }
    public string Message { get; set; }
}

public class Logger
{
    public void Info(string message) => Console.WriteLine($"INFO: {message}");
    public void Error(string message) => Console.WriteLine($"ERROR: {message}");
}

public class HardwareFingerprint
{
    public string GenerateFingerprint()
    {
        // 生成硬件指纹的逻辑
        return "dummy_fingerprint";
    }
}

File: examples/Java/LicenseValidator.java
import javax.crypto.Cipher;
import javax.crypto.spec.SecretKeySpec;
import java.nio.charset.StandardCharsets;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.Base64;
import org.json.JSONObject;

public class LicenseValidator {
    private final Logger logger;
    private final HardwareFingerprint fingerprint;

    public LicenseValidator() {
        this.logger = new Logger();
        this.fingerprint = new HardwareFingerprint();
    }

    public ValidationResult validateLicense(String licenseFile, String licenseKey, boolean validateFootprint) {
        try {
            // 加载并解密许可证文件
            String encryptedLicense = new String(Files.readAllBytes(Paths.get(licenseFile)), StandardCharsets.UTF_8);
            if (encryptedLicense.isEmpty()) {
                return validationFailed("无法加载许可证文件");
            }

            String decryptedLicense = decrypt(encryptedLicense, licenseKey);
            JSONObject licenseData = new JSONObject(decryptedLicense);

            // 生成当前硬件指纹
            String currentHardwareFingerprint = fingerprint.generateFingerprint();
            if (validateFootprint && !licenseData.getString("hardware_fingerprint").equals(currentHardwareFingerprint)) {
                return validationFailed("硬件指纹不匹配");
            }

            // 检查许可证是否过期
            long currentTime = System.currentTimeMillis() / 1000L;
            long expiresAt = licenseData.getLong("expires_at");
            long issuedAt = licenseData.getLong("issued_at");

            if (currentTime > expiresAt || currentTime < issuedAt) {
                return validationFailed("许可证已过期");
            }

            ValidationResult result = new ValidationResult(true, expiresAt, licenseData.getJSONArray("features").toList().toArray(new String[0]), licenseData.getInt("max_users"));
            logger.info("许可证验证成功，详情: " + result);
            return result;
        } catch (Exception e) {
            logger.error("许可证验证错误: " + e.getMessage());
            return validationFailed(e.getMessage());
        }
    }

    private String decrypt(String encryptedText, String key) throws Exception {
        // 解密逻辑
        return "dummy_decrypted_license";
    }

    private ValidationResult validationFailed(String message) {
        return new ValidationResult(false, 0, new String[0], 0, message);
    }

    public static void main(String[] args) {
        LicenseValidator validator = new LicenseValidator();
        ValidationResult result = validator.validateLicense("path/to/license/file", "your_license_key", true);
        System.out.println(result);
    }
}

class ValidationResult {
    private boolean valid;
    private long expiry;
    private String[] features;
    private int maxUsers;
    private String message;

    public ValidationResult(boolean valid, long expiry, String[] features, int maxUsers) {
        this.valid = valid;
        this.expiry = expiry;
        this.features = features;
        this.maxUsers = maxUsers;
    }

    public ValidationResult(boolean valid, long expiry, String[] features, int maxUsers, String message) {
        this.valid = valid;
        this.expiry = expiry;
        this.features = features;
        this.maxUsers = maxUsers;
        this.message = message;
    }

    @Override
    public String toString() {
        return "ValidationResult{" +
                "valid=" + valid +
                ", expiry=" + expiry +
                ", features=" + String.join(", ", features) +
                ", maxUsers=" + maxUsers +
                ", message='" + message + '\'' +
                '}';
    }
}

class Logger {
    public void info(String message) {
        System.out.println("INFO: " + message);
    }

    public void error(String message) {
        System.err.println("ERROR: " + message);
    }
}

class HardwareFingerprint {
    public String generateFingerprint() {
        // 生成硬件指纹的逻辑
        return "dummy_fingerprint";
    }
}